Public/Src/Cache/VerticalStore/MemoizationStoreAdapter/MemoizationStoreCacheFactory.cs

```diff
@@ -357,33 +356,37 @@ private static void SetDefaultsForStreamCas(Config config)
             }
         }
 
-        private static SQLiteMemoizationStoreConfiguration GetMemoConfig(AbsolutePath cacheRoot, Config config, CasConfig configCore)
+        private static MemoizationStoreConfiguration GetMemoizationStoreConfiguration(AbsolutePath cacheRoot, Config config, CasConfig configCore)
         {
-            var memoConfig = new SQLiteMemoizationStoreConfiguration(cacheRoot)
+            if (config.UseRocksDbMemoizationStore)
+            {
+                var rocksDbConfig = new RocksDbMemoizationStoreConfiguration() {
+                    Database = new RocksDbContentLocationDatabaseConfiguration(cacheRoot / "rocksdb"),
+                };
+
+                return rocksDbConfig;
+            }
+
+            var sqliteConfig = new SQLiteMemoizationStoreConfiguration(cacheRoot)

```

Comment:
Minor: I'm a fan of early return where these is an error case or trivial case being tested for and the early return then allows the "meat" of the implementation to not be buried 6 brackets deep. But in this case the two cases have approximately equal weight, so I would argue for "else" here.